packages/simcore-sdk/src/simcore_sdk/node_ports_common/aws_s3_cli_utils.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import re
+
+from pydantic import ByteSize, parse_obj_as
+from servicelib.logging_utils import log_catch
+from servicelib.progress_bar import ProgressBarData
+
+from .r_clone_utils import BaseLogParser
+
+_logger = logging.getLogger(__name__)
+
+
+def _parse_size(log_string):
+    match = re.search(r"(\d+\.\d+|\d+)\s*(GiB|MiB|KiB|TiB|B)", log_string)
+    if match:
+        value = float(match.group(1))
+        unit = match.group(2)
+        return value, unit
+    return None, None
+
```

Comment:
I think here you might be much better off using
```python
try:
    return parse_obj_as(ByteSize, log_string)
except ValidationError
    return None
```
This understands all the units and can convert into human readable and is in the end an integer in byte unit.